rotkehlchen/tests/api/test_counterparty_asset_mappings.py

```diff
@@ -0,0 +1,270 @@
+        ),
+        json={'counterparty': 'hyperliquid'},
+    )
+    result = assert_proper_sync_response_with_result(response)
+    assert len(result['entries']) == result['entries_found'] == 0
+
+    # query by symbol all the kraken mappings
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'counterpartyassetmappingsresource',
+        ),
+        json={'counterparty_symbol': 'HYPE', 'counterparty': 'hyperliquid'},
+    )
+    result = assert_proper_sync_response_with_result(response)
+    assert all(entry['counterparty'] == 'hyperliquid' for entry in result['entries'])
+
+
+def test_counterparty_asset_mappings_add(rotkehlchen_api_server: 'APIServer') -> None:
+    """Test the counterparty asset mappings API for adding the mappings."""
+    all_mappings = _get_all_counterparty_mappings(rotkehlchen_api_server)['entries']
+    added_mappings = [{
+        'asset': 'eip155:1/erc20:0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984',
+        'counterparty_symbol': 'NEW1',
+        'counterparty': 'hyperliquid',
+    }, {
+        'asset': 'eip155:1/erc20:0xF57e7e7C23978C3cAEC3C3548E3D615c346e79fF',
+        'counterparty_symbol': 'NEW2',
+        'counterparty': 'hyperliquid',
+    }]
+
+    # add two new mappings
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'counterpartyassetmappingsresource',
+        ),
+        json={
+            'entries': added_mappings,
+        },
+    )
+    result = assert_proper_sync_response_with_result(response)
+    assert result is True
+
+    # check that the mappings were indeed added
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'counterpartyassetmappingsresource',
+        ),
+        json={
+            'offset': 0,
+            'limit': 4000,
+        },
+    )
+    result = assert_proper_sync_response_with_result(response)
+    all_mappings_after_addition = result['entries']
+    assert len(all_mappings_after_addition) == result['entries_found'] == NUM_PACKAGED_COUNTERPARTY_ASSETS_MAPPINGS + 2  # noqa: E501
+    for new_mapping in added_mappings:
+        assert new_mapping not in all_mappings
+        assert new_mapping in all_mappings_after_addition
+
+
+def test_counterparty_asset_mappings_update(rotkehlchen_api_server: 'APIServer') -> None:
+    """Test the counterparty asset mappings API for updating the mappings."""
+    assert_proper_sync_response_with_result(requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'counterpartyassetmappingsresource',
+        ),
+        json={
+            'entries': [{
+                'asset': 'eip155:1/erc20:0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984',
+                'counterparty_symbol': 'NEW1',
+                'counterparty': 'hyperliquid',
+            }],
+        },
+    ))
+    all_mappings = _get_all_counterparty_mappings(rotkehlchen_api_server)['entries']
```

Comment:
don't do it by querying the api, query using the functions that you have underneath. It is using more logic that you don't really need. Same for all the other places that you use `entries`. It is good that you test the endpoint but is not needed if you just want to fetch them